runners/flink/src/test/java/org/apache/beam/runners/flink/FlinkTransformOverridesTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.flink;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+
+import java.util.Collections;
+import javax.annotation.Nullable;
+import org.apache.beam.runners.flink.FlinkStreamingPipelineTranslator.StreamingShardedWriteFactory;
+import org.apache.beam.sdk.coders.VoidCoder;
+import org.apache.beam.sdk.io.DynamicFileDestinations;
+import org.apache.beam.sdk.io.FileBasedSink;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.io.WriteFiles;
+import org.apache.beam.sdk.io.WriteFilesResult;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.SerializableFunctions;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.transforms.windowing.PaneInfo;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * Tests if overrides are properly applied.
+ */
+public class FlinkTransformOverridesTest {
+
+    @Rule
+    public transient TemporaryFolder tmpFolder = new TemporaryFolder();
+
+    @Test
+    public void testDefaultParallelism() {
+        FlinkPipelineOptions options = PipelineOptionsFactory.as(FlinkPipelineOptions.class);
+        options.setRunner(TestFlinkRunner.class);
+        options.setFlinkMaster("[auto]");
+
+        testStreamingWriteOverride(options, 2);
```

Comment:
A default of 2 seems low to me; it might be worth commenting where it's set about the default amount and why it's sensible.

Reply:
Ah, it's the default parallelism, which is 1, times 2. Which is also used for Dataflow. Might make sense to use some larger value if `parallelism * 2` is too low. What do you think?

Reply:
I'm not hugely concerned, especially if we don't get parallelism unless configured. I'd like a comment at the site where it's set, and probably a `checkArgument` (if we start using something like 0 or -1 to signal "runner-determined", I don't know what'll happen if we set 0 shards)

Reply:
Would set the parallelism explicitly here on the options. This should fail in local mode where the default parallelism now depends on the number of available cores.